wrappers/android/tools/camera/src/main/java/com/intel/realsense/camera/SettingsActivity.java

```diff
@@ -208,9 +214,16 @@ public boolean  onChildClick(ExpandableListView parent, View v,
                         break;
                     }
                     case INDEX_UPDATE_UNSIGNED: {
-                        Intent intent = new Intent(SettingsActivity.this, FileBrowserActivity.class);
-                        intent.putExtra(getString(R.string.browse_folder), getString(R.string.realsense_folder) + File.separator +  "firmware");
-                        startActivityForResult(intent, OPEN_FILE_REQUEST_CODE);
+                        Intent intent = new Intent();
+                        intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION | Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
```

Comment:
Please explain why do you need these permissions

Reply:
these are permissions for the uri, probably not needed as it's in the same app. removed it.